lib/matplotlib/image.py

```diff
@@ -419,14 +428,18 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
                 # colormapping works correctly
                 hid_output = output
                 output = np.ma.masked_array(
-                    hid_output[..., 0], hid_output[..., 3] < 0.5)
+                    hid_output[..., 0], hid_output[..., 3] == 0)
```

Comment:
Hm, but this causes trouble if there are *bad* data in image. For example, bad values in one of the tests are shown as blue. If we restore the alpha channel, all bad blue values will become transparent. As a compromise, I have rerendered the problematic images (changes only occur at the boundary) and I only restore the alpha channel if `_array_alpha` is given.